nanoFramework.Json/TimeExtensions.cs

```diff
@@ -56,13 +56,23 @@ public static DateTime FromiCalendar(string iCalendar)
 
 			// We now have the time string to parse, and we'll adjust
 			// to UTC or timezone after parsing
-			string year = time.Substring(0, 4);
-			string month = time.Substring(4, 2);
-			string day = time.Substring(6, 2);
-			string hour = time.Substring(9, 2);
+			string year   = time.Substring(0, 4);
```

Comment:
please us ctrl + k +d to make sure code is properly aligned. The new code is not aligned, has extra spaces, missing some enter.

Reply:
Done.
Sorry, I learned C in the early 80's and still prefer that style.
Cleaned up a couple other places too.

Reply:
Just pushed changes to clean the code up.
Please take a look and let me know if I missed anything.
Apparently, my original performance testing must have somehow compared the Release version to the Debug version of the old code (not sure how this happened).

I ran the new code (in Release mode) against the unmodified library (also in Release mode) and the numbers were almost the same.
The numbers are also very close with the same test done in Debug mode.
(Debug mode takes longer for obvious reasons)

Looks like we can't claim a performance benefit with the validation changes in this PR.
But that also means that the Native exceptions are not causing delays.